plugins/content/createmenulink/createmenulink.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Content.createmenulink
+ *
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\Registry\Registry;
+
+JLoader::import('joomla.application.component.model');
+
+/**
+ * Createmenulink Plugin
+ *
+ * @since  3.6
+ */
+class PlgContentCreatemenulink extends JPlugin
+{
+	/**
+	 * Load the language file on instantiation.
+	 *
+	 * @var    boolean
+	 * @since  3.6
+	 */
+	protected $autoloadLanguage = true;
+
+	/**
+	 * Runs on content preparation
+	 *
+	 * @param   string  $context  The context for the data
+	 * @param   object  $data     An object containing the data for the form.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+	public function onContentPrepareData($context, $data)
+	{
+		if (!in_array($context, array('com_content.article','com_contact.contact')))
+		{
+			return true;
+		}
+
+		$app = JFactory::getApplication();
+
+		if ($app->isAdmin())
```

Comment:
You should make a negative check here and return true if !admin.

No reason to have the whole block in an if case.

```php
if (!$app->isAdmin())
{
      return true;
} 
```